nav2_costmap_2d/include/nav2_costmap_2d/layer.hpp

```diff
@@ -64,7 +65,8 @@ class Layer  // TODO(mjeronimo): public nav2_util::LifecycleHelperInterface
     tf2_ros::Buffer * tf,
     nav2_util::LifecycleNode::SharedPtr node,
     rclcpp::Node::SharedPtr client_node,
-    rclcpp::Node::SharedPtr rclcpp_node);
+    rclcpp::Node::SharedPtr rclcpp_node,
+    nav2_util::ParameterEventsSubscriber * param_subscriber = nullptr);
```

Comment:
Why not a shared ptr?

Is there a performance reason to have this shared with all the layers?

Reply:
This could be a shared ptr. 

I had decided to make the subscriber object shared between layers because since they are on the same node, I didn't think they needed new objects which would then create new subscriptions (the class creates new subscriptions for unique nodes/namespaces parameter events). With sharing, there need only be one subscription created (on the node itself).

Reply:
Ok. Can we make it shared?